### PR TITLE
Add Pop-up on save click for temporary notebooks

### PR DESCRIPTION
**Pull request by kcheekuri**
[Preview this branch](https://ms.portal.azure.com/?feature.phoenix=true&feature.junoendpoint=https://juno-test.documents-dev.windows-int.net&dataExplorerSource=https://cosmos-explorer-preview.azurewebsites.net/commit/12e0de7d2df8ab418acea1ae35929d83875c38bd/explorer.html?feature.pr=https://github.com/Azure/cosmos-explorer/pull/1177#home)

This PR:

- Add Pop-up on save click for temporary notebooks.
- Adds a prompt message when saving a notebook to temporary workspace only and not for github.

### CODE REVIEW

**Review by ramrajprabu — DISMISSED** [review]

File: src/Explorer/Tabs/NotebookTabBase.ts [hunk 1]
```diff
@@ -45,6 +50,20 @@ export default class NotebookTabBase extends TabsBase {
...
     }
+    if (
+      options.notebookContentItem &&
+      NotebookUtil.getContentProviderType(options.notebookContentItem.path) ===
+        NotebookContentProviderType.JupyterContentProviderType
+    ) {
+      // There is no way to turn off auto-save, set to 1 year
+      NotebookTabBase.clientManager
+        ?.getStore()
+        .dispatch(configOption("autoSaveInterval").action(365 * 24 * 3600 * 1000));
```
Comment (gitanuj) [inline src/Explorer/Tabs/NotebookTabBase.ts:61]: Are we using different stores for every tab?

Comment (gitanuj) [reply src/Explorer/Tabs/NotebookTabBase.ts:61]: autoSaveInternal is a setting on store, and looks like we share the store so this setting is global for all the tabs. Can you please test if it works locally by setting the interval to like 5s?

File: src/Common/Constants.ts [hunk 2]
```diff
@@ -378,7 +378,7 @@ export class Notebook {
     "We have identified an issue with the Cassandra Shell and it is unavailable right now. We are actively working on the mitigation.";
   public static saveNotebookModalTitle = "Save notebook in temporary workspace";
   public static saveNotebookModalContent =
-    "This notebook will be saved in the temporary workspace and will be removed when the session expires. To save your work permanently, save your notebooks to a GitHub repository or download the notebooks to your local machine before the session ends.";
```
Comment (srinathnarayanan) [inline src/Common/Constants.ts:381]: why are we removing this?

Comment (kcheekuri) [reply src/Common/Constants.ts:381]: Spliting the text for showing it in next paragragh

File: src/Explorer/Tabs/NotebookTabBase.ts (hunk 1 above)
Comment (kcheekuri) [reply src/Explorer/Tabs/NotebookTabBase.ts]: Removed this in this PR

File: src/Explorer/Notebook/NotebookClientV2.ts [hunk 3]
```diff
@@ -243,10 +243,10 @@ export class NotebookClientV2 {
 
     // Additional configuration
     this.store.dispatch(configOption("editorType").action(params.cellEditorType ?? "codemirror"));
+    this.store.dispatch(configOption("codeMirror.lineNumbers").action(true));
```
Comment (srinathnarayanan) [inline src/Explorer/Notebook/NotebookClientV2.ts:246]: why was this added?

Comment (kcheekuri) [reply src/Explorer/Notebook/NotebookClientV2.ts:246]: It just order changed

**Review by srinathnarayanan — APPROVED** [review]

**Review by gitanuj — APPROVED** [review]
